gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/BloodPressureMapperTest.java

```diff
@@ -0,0 +1,201 @@
+            .isEqualTo(COMPONENT_1_VALUE_QUANTITY_VALUE);
+        assertThat(bloodPressure.getComponent().get(0).getInterpretation().getText()).isEqualTo(COMPONENT_1_INTERPRETATION_TEXT);
+        assertThat(bloodPressure.getComponent().get(0).getReferenceRange().get(0).getText())
+            .isEqualTo(COMPONENT_1_REFERENCE_RANGE_TEXT);
+
+        assertThat(bloodPressure.getComponent().get(1).getCode().getCodingFirstRep().getDisplay()).isEqualTo(CODING_DISPLAY_MOCK);
+        assertThat(bloodPressure.getComponent().get(1).getValueQuantity().getValue().toString())
+            .isEqualTo(COMPONENT_2_VALUE_QUANTITY_VALUE);
+        assertThat(bloodPressure.getComponent().get(1).getInterpretation().getText()).isEqualTo(COMPONENT_2_INTERPRETATION_TEXT);
+        assertThat(bloodPressure.getComponent().get(1).getReferenceRange().get(0).getText()).isEqualTo(COMPONENT_2_REFERENCE_RANGE_TEXT);
+    }
+
+    @Test
+    public void mapBloodPressureWithNoOptionalData() {
+        when(codeableConceptMapper.mapToCodeableConcept(any())).thenReturn(CODEABLE_CONCEPT);
+
+        var ehrExtract = unmarshallEhrExtractElement("no_optional_data_bp_example.xml");
+
+        var bloodPressure = bloodPressureMapper.mapBloodPressure(ehrExtract, patient, ENCOUNTER_LIST).get(0);
+
+        assertFixedValues(bloodPressure);
+        assertThat(bloodPressure.getId()).isEqualTo(EXAMPLE_ID);
+        assertThat(bloodPressure.getCode().getCodingFirstRep().getDisplay()).isEqualTo(CODING_DISPLAY_MOCK);
+        assertThat(bloodPressure.hasSubject()).isTrue();
+        assertThat(bloodPressure.getIssuedElement().asStringValue()).isEqualTo(ISSUED_EXAMPLE);
+        assertThat(bloodPressure.getPerformer().get(0).getReference()).isEqualTo(PPRF_PARTICIPANT_ID);
+
+        assertThat(bloodPressure.getEffective()).isNull();
+        assertThat(StringUtils.isEmpty(bloodPressure.getComment()));
+
+        assertThat(bloodPressure.getComponent().get(0).getCode().getCodingFirstRep().getDisplay()).isEqualTo(CODING_DISPLAY_MOCK);
+        assertThat(bloodPressure.getComponent().get(0).getValueQuantity()).isNull();
+        assertThat(bloodPressure.getComponent().get(0).getInterpretation().getCoding().isEmpty());
+        assertThat(bloodPressure.getComponent().get(0).getReferenceRange().isEmpty());
+
+        assertThat(bloodPressure.getComponent().get(1).getCode().getCodingFirstRep().getDisplay()).isEqualTo(CODING_DISPLAY_MOCK);
+        assertThat(bloodPressure.getComponent().get(1).getValueQuantity()).isNull();
+        assertThat(bloodPressure.getComponent().get(1).getInterpretation().getCoding().isEmpty());
+        assertThat(bloodPressure.getComponent().get(1).getReferenceRange().isEmpty());
+    }
+
+    @Test
+    public void mapBloodPressureObservationWithCompositionIdMatchingEncounter() {
+        var ehrExtract = unmarshallEhrExtractElement(
+            "ehr_composition_id_matching_encounter_bp_example.xml");
```

Comment:
I think all of the file names can still fit in one line with unmarshallEhrExtractElement (just saying :D)

Reply:
I think a few are still too long but I've fixed the rest :)